ports/geckolib/wrapper.rs

```diff
@@ -95,16 +95,40 @@ impl<'ln> GeckoNode<'ln> {
     }
 }
 
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, Debug)]
 pub struct GeckoRestyleDamage(nsChangeHint);
 
 impl TRestyleDamage for GeckoRestyleDamage {
     type PreExistingComputedValues = nsStyleContext;
+
+    fn empty() -> Self {
+        use std::mem;
+        GeckoRestyleDamage(unsafe { mem::transmute(0u32) })
+    }
+
     fn compute(source: Option<&nsStyleContext>,
                new_style: &Arc<ComputedValues>) -> Self {
```

Comment:
Per discussion, let's make this take a Non-Option, and then the caller can make stronger assertions about the reason why the style context doesn't exist (i.e. either non-incremental restyle or the display::none case).